src/v2/providers/https.ts

```diff
@@ -493,3 +494,94 @@ export function onCall<T = any, Return = any | Promise<any>, Stream = unknown>(
   };
   return func;
 }
+
+// To avoid taking a strict dependency on Genkit we will redefine the limited portion of the interface we depend upon.
+// A unit test (dev dependency) notifies us of breaking changes.
+interface ZodType<T = any> {
+  __output: T;
+}
+
+interface GenkitRunOptions {
+  context?: any;
+}
+
+type GenkitAction<
+  I extends ZodType = ZodType<any>,
+  O extends ZodType = ZodType<any>,
+  S extends ZodType = ZodType<any>
+> = {
+  // NOTE: The return type from run includes trace data that we may one day like to use.
+  run(input: I["__output"], options: GenkitRunOptions): Promise<{ result: O["__output"] }>;
+  stream(
+    input: I["__output"],
+    options: GenkitRunOptions
+  ): { stream: AsyncIterable<S["__output"]>; output: Promise<O["__output"]> };
+
+  __action: {
+    name: string;
+  };
+};
+
+type ActionInput<F extends GenkitAction> = F extends GenkitAction<infer I extends ZodType, any, any>
+  ? I["__output"]
+  : never;
+type ActionOutput<F extends GenkitAction> = F extends GenkitAction<
+  any,
+  infer O extends ZodType,
+  any
+>
+  ? O["__output"]
+  : never;
+type ActionStream<F extends GenkitAction> = F extends GenkitAction<
+  any,
+  any,
+  infer S extends ZodType
+>
+  ? S["__output"]
+  : never;
+
+export function onCallGenkit<A extends GenkitAction>(
+  action: A
+): CallableFunction<ActionInput<A>, Promise<ActionOutput<A>>, ActionStream<A>>;
+export function onCallGenkit<A extends GenkitAction>(
+  opts: CallableOptions<ActionInput<A>>,
+  flow: A
+): CallableFunction<ActionInput<A>, Promise<ActionOutput<A>>, ActionStream<A>>;
+export function onCallGenkit<A extends GenkitAction>(
+  optsOrAction: A | CallableOptions<ActionInput<A>>,
+  action?: A
+): CallableFunction<ActionInput<A>, Promise<ActionOutput<A>>, ActionStream<A>> {
+  let opts: CallableOptions<ActionInput<A>>;
+  if (arguments.length === 2) {
+    opts = optsOrAction as CallableOptions<ActionInput<A>>;
+  } else {
+    opts = {};
+    action = optsOrAction as A;
+  }
+  if (!opts.secrets?.length) {
+    logger.debug(
```

Comment:
q: is this visible during `firebase deploy`?

Reply:
Only if you do --debug I think. Can figure out for sure later, but I have to run and want to let you cut a release so I can build upon it over the weekend.

Reply:
Alternatively, we could push this into the CLI where it notices a genkit onCall with no secrets?